build.proj

```diff
@@ -76,9 +83,31 @@
           CustomErrorRegularExpression="(^Unable to locate .*)|(^Updating the invalid lock file with .*)"
           ContinueOnError="ErrorAndContinue" />
 
+    <!-- Given we ErrorAndContinue we need to propagate the error if the overall task failed -->
+    <Error Condition="'$(MSBuildLastTaskResult)'=='false'" />
+    
     <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Restoring all packages...Done." />
   </Target>
 
+  <!-- Evaluate our test projects (in src\tests.builds) -->
+  <Target Name="BatchGenerateTestProjectJsons"
+          Condition="'$(BuildTestsAgainstPackages)' == 'true'" 
+          DependsOnTargets="FilterProjects"
+          BeforeTargets="RestorePackages">
+      <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Generating Test project.json's..." />
+      <MSBuild Targets="GenerateAllTestProjectJsons"
+               Projects="src\tests.builds" />
+      <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Generating Test project.json's...done" />
+  </Target>
```

Comment:
Question only: this means that if I do build.cmd /p:BuildTestsAgainstPackages=true, then i'll also be testing just against whatever published packages are? 

It seems like we pulled in only part of this change - see https://github.com/dotnet/corefx/commit/c1ec94b74183f598513199cde0d799a514c0a0fb. Do we need to pull the changes in dir.props as well?